security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -749,6 +754,8 @@ func (sc *SecretManagerClient) UpdateConfigTrustBundle(trustBundle []byte) error
 	sc.configTrustBundleMutex.Unlock()
 	cacheLog.Debugf("update new trust bundle")
 	sc.OnSecretUpdate(security.RootCertReqResourceName)
+	sc.cache.SetWorkload(nil)
+	sc.OnSecretUpdate(security.WorkloadKeyCertResourceName)
```

Comment:
@howardjohn as to `Typically root cert migrations add the new trust for quite a while before using it IME`

You can take a look, rootcert is always pushed beforehand, then workload cert.  I believe this is not anything wrong

Reply:
Could one workload obtain the new roots and workloads cert before another workload has the new roots? This is where is could see this causing downtime.